contracts/delegatecall/LibEtherForENS.sol

```diff
@@ -0,0 +1,25 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.20;
+
+abstract contract Resolver {
+    function addr(bytes32 node) public view virtual returns (address);
+}
+
+abstract contract ENS {
+    function resolver(bytes32 node) public view virtual returns (Resolver);
+}
+
+library LibEtherForENS {
+    function etherForENS(bytes32 node, uint256 value, bytes memory payload) external{
```

Comment:
Given that we're allowing a `payload` to be passed, this is more like an arbitrary low-level call to an address through its ENS name. Maybe rename this library to `LibCallToENS`?